src/controllers/auth.js

```diff
@@ -0,0 +1,112 @@
+import db from "../db/index";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcryptjs";
+
+class Authentication {
+
+    static async login(req, res, next) {
+        const { username, password } = req.body;
+
+        try {
+            if (!username || !password) {
+                return res.status(400).json({
+                    message: 'all fields required',
+                    code: 400
+                })
+            }
+
+            const userExistQuery = 'SELECT * FROM users WHERE username=$1';
+            const userExistValue = [username]
+            const userExist = await db.query(userExistQuery, userExistValue);
+            if (userExist.rows[0]) {
+                const match = await bcrypt.compare(
+                    password,
+                    userExist.rows[0].password,
+                );
+
+                if (match) {
+                    return jwt.sign({
+                        username: userExist.rows.username,
+                        id: userExist.rows.id
+                    }, process.env.SECRET_JWT_KET, { expiresIn: '7d' }, function (err, token) {
+                        if (err) {
+                            return res.status(403).send(err);
+                        } else {
+                            return res.status(200).json({
+                                message: "login success",
+                                token: token
+                            })
+                        }
+                    })
+                } else {
+                    return res.status(403).json({
+                        message: "Incorrect username and password",
+                        code: 403
+                    })
+                }
+            }
+
+            return res.status(403).json({
+                message: "Incorrect username and password",
+                code: 403
+            })
+        } catch (error) {
+            next(error);
+        }
+    }
+
+
+
+
+    static async signup(req, res, next) {
+        const { username,
+            password,
+            firstname,
+            lastname,
+            email
+        } = req.body;
+
+        try {
+            if (!username || !password) {
```

Comment:
this is good.

Also do character and length validation. As it is, people can user spaces their username and it will work.
There's a tool called `joi` for validation. check it out on npm

Reply:
okay, I have used hapi/joi before, I will integrate it for validation